force-app/main/default/classes/CourseHandler.cls

```diff
@@ -60,4 +60,23 @@ global with sharing class CourseHandler {
             JSON.serialize(wrapperList)
         );
     }
+
+    public override void onBeforeUpdate(Map<Id, sObject> triggerOldMap) {
+        List<Course__c> canceledCourses = new List<Course__c>();
+        for (Course__c course : (List<Course__c>) records) {
+            if (course.Cancel__c != false) {
+                canceledCourses.add(course);
+            }
+        }
+        if (canceledCourses.size() > 0) {
+            changeCourseTitle(canceledCourses);
+        }
+    }
+
+    public static void changeCourseTitle(List<Course__c> canceledCourses) {
+        for (Course__c course : canceledCourses) {
+            course.Name = 'AVLYST' + course.Name;
+            course.Active__c = false;
+        }
+    }
```

Comment:
All logikken for triggeren må skilles ut og ligge i en egen klasse. Se eksempel i klassen Contact_Create eller Course_AddUrl.